### PR TITLE
feature: Adapt to manifest v3

### PR DESCRIPTION
**Pull request by doosuu**
# Description

This PR adapts the template to use the new app manifest v3 structure.

## Checklist

* [x] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [x] Vehicle App can process MQTT messages and call the seat service
* [x] Vehicle App can be deployed to local K3D and is running
* [x] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [ ] Extended the documentation in Velocitas repo
* [ ] Extended the documentation in README.md

* [x] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully

* [ ] Release workflow is passing

### CODE REVIEW

File: .github/workflows/build-docker-image.yml [hunk 1]
```diff
@@ -21,30 +21,17 @@ on:
...
         type: string
-      multiarch:
-        required: false
-        type: boolean
-        default: false
-    outputs:
-      imageid:
-        value: ${{ jobs.build-image.outputs.imageid }}
-      imagedigest:
-        value: ${{ jobs.build-image.outputs.digest }}
```
Comment (BjoernAtBosch) [inline .github/workflows/build-docker-image.yml:35]: Why removing this? Not needed any more?

File: app/AppManifest.json [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
+                    "required": [
+                        {
+                            "path": "Vehicle.Speed",
+                            "required": "true",
+                            "access": "read"
+                        },
+                        {
+                            "path": "Vehicle.Cabin.Seat.Row1.Pos1.Position",
+                            "required": "true",
+                            "access": "write"
```
Comment (BjoernAtBosch) [inline app/AppManifest.json]: Actually not used - remove:
```suggestion
```

File: app/src/SampleApp.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace example {
+
+const auto GET_SPEED_REQUEST_TOPIC       = "seatadjuster/setPosition/request";
+const auto GET_SPEED_RESPONSE_TOPIC      = "seatadjuster/setPosition/response";
+const auto DATABROKER_SUBSCRIPTION_TOPIC = "seatadjuster/currentPosition";
+
+SampleApp::SampleApp()
+    : VehicleApp(velocitas::IVehicleDataBrokerClient::createInstance("vehicledatabroker"),
+                 velocitas::IPubSubClient::createInstance("SeatAdjusterApp")) {}
```
Comment (BjoernAtBosch) [inline app/src/SampleApp.cpp]:
```suggestion
                 velocitas::IPubSubClient::createInstance("SampleApp")) {}
```

File: app/src/SampleApp.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+#include <fmt/core.h>
+#include <nlohmann/json.hpp>
+#include <utility>
+
+namespace example {
+
+const auto GET_SPEED_REQUEST_TOPIC       = "seatadjuster/setPosition/request";
+const auto GET_SPEED_RESPONSE_TOPIC      = "seatadjuster/setPosition/response";
+const auto DATABROKER_SUBSCRIPTION_TOPIC = "seatadjuster/currentPosition";
```
Comment (BjoernAtBosch) [inline app/src/SampleApp.cpp]:
```suggestion
const auto GET_SPEED_REQUEST_TOPIC       = "sampleapp/getSpeed";
const auto GET_SPEED_RESPONSE_TOPIC      = sampleapp/getSpeed/response";
const auto DATABROKER_SUBSCRIPTION_TOPIC = "sampleapp/currentSpeed";
```

File: app/AppManifest.json [hunk 5]
```diff
@@ -1,26 +1,37 @@
...
-        ]
-    }
-]
+        {
+            "type": "pubsub",
+            "config": {
+                "reads": [ "seatadjuster/setPosition/request" ],
+                "writes": [
+                    "seatadjuster/setPosition/response",
+                    "seatadjuster/currentPosition"
```
Comment (BjoernAtBosch) [inline app/AppManifest.json]:
```suggestion
                "reads": [ "sampleapp/getSpeed" ],
                "writes": [
                    "sampleapp/getSpeed/response",
                    "sampleapp/currentSpeed"
```

**Review by BjoernAtBosch — CHANGES_REQUESTED** [review]
One question and some proposed fixes

File: .github/workflows/build-docker-image.yml (hunk 1 above)
Comment (doosuu) [reply .github/workflows/build-docker-image.yml:35]: @MP91 can you elaborate why this was removed?

Comment (MP91) [reply .github/workflows/build-docker-image.yml:35]: Because I did not saw any place where it is used. Did I miss something?

Comment (doosuu) [reply .github/workflows/build-docker-image.yml:35]: Looks good from my side, the output variables were never used, where they @MP91 ?

**Review by BjoernAtBosch — APPROVED** [review]
LGTM

File: .github/workflows/build-docker-image.yml (hunk 1 above)
Comment (MP91) [reply .github/workflows/build-docker-image.yml:35]: I thought we were talking just about that :D But no I did not find any place were we used it...

**Pull request merged**